CONTRIBUTING.md

```diff
@@ -25,13 +26,76 @@ _As contributors and maintainers of this project, and in the interest of fosteri
 
 ## Getting Started
 
-Dev guides:
-
-- [Contribution Guide]
-- [MacOS Dev Guide]
-- [Windows Dev Guide]
-
-General resources for contributors:
+### Forking Kustomize and Working Locally
+The Kustomize project uses a "Fork and Pull" workflow that is standard to GitHub. In git terms, your personal fork is referred to as the "origin" and the actual project's git repository is called "upstream". To keep your personal branch (origin) up to date with the project (upstream), it must be configured within your local working copy.
+
+### Create a fork in GitHub
+1. Visit https://github.com/kubernetes-sigs/kustomize
+2. Click the `Fork` button on the top right
+
+### Clone the repository
+```bash
+# Clone your repository fork from the previous step
+export user=<your github profile name>
+git clone --recurse-submodules git@github.com:$user/kustomize.git
+cd kustomize
+
+# Configure upstream
+git remote add upstream https://github.com/kubernetes-sigs/kustomize
+git remote set-url --push upstream no_push
+
+# Review git configuration
+git remote -v
+```
+
+### Create a working branch
+You will need to periodically fetch changes from the `upstream` repository to keep your working branch in sync.
+```bash
+# Fetch changes from upstream master
+cd kustomize
+git fetch upstream
+git checkout master
+git rebase upstream/master
+
+# Create your working branch
+git checkout -b myfeature
+```
+
+### Push to GitHub
+When your changes are ready for review, push your working branch to your fork on GitHub.
+```bash
+cd kustomize
+export user=<your github profile name>
+git push git@github.com:$user/kustomize.git myfeature
+```
+
+### Create a Pull Request
+1. Visit your fork at `https://github.com/<user>/kustomize`
+2. Click the **Compare & Pull Request** button next to your `myfeature` branch.
+3. Check out the pull request [process](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md) for more details and advice.
+
```

Comment:
You could also call out that if people are using `git` from the command line, GitHub responds to the CLI with a handy link to create a PR that can be used for this purpose.